File: src/spacetraders/application/player/queries/list_players.py
from dataclasses import dataclass
from typing import List
from pymediatr import Request, RequestHandler

from ....domain.shared.player import Player
from ....ports.repositories import IPlayerRepository

@dataclass(frozen=True)
class ListPlayersQuery(Request[List[Player]]):
    """Query to list all players"""
    pass

class ListPlayersHandler(RequestHandler[ListPlayersQuery, List[Player]]):
    """Handler for listing all players"""

    def __init__(self, player_repository: IPlayerRepository):
        self._player_repo = player_repository

    async def handle(self, request: ListPlayersQuery) -> List[Player]:
        """List all registered players"""
        return self._player_repo.list_all()

File: gobot/internal/application/player/list_players.go
package player

import (
	"context"
	"fmt"

	"github.com/andrescamacho/spacetraders-go/internal/application/common"
	"github.com/andrescamacho/spacetraders-go/internal/domain/player"
)

// ListPlayersCommand represents a command to list all players
type ListPlayersCommand struct {
	// No parameters - lists all players
}

// ListPlayersResponse represents the result of listing players
type ListPlayersResponse struct {
	Players []*player.Player
}

// ListPlayersHandler handles the ListPlayers command
type ListPlayersHandler struct {
	playerRepo player.PlayerRepository
}

// NewListPlayersHandler creates a new ListPlayersHandler
func NewListPlayersHandler(playerRepo player.PlayerRepository) *ListPlayersHandler {
	return &ListPlayersHandler{
		playerRepo: playerRepo,
	}
}

// Handle executes the ListPlayers command
func (h *ListPlayersHandler) Handle(ctx context.Context, request common.Request) (common.Response, error) {
	_, ok := request.(*ListPlayersCommand)
	if !ok {
		return nil, fmt.Errorf("invalid request type: expected *ListPlayersCommand")
	}

	// TODO: Add ListAll method to PlayerRepository
	// For now, return empty list
	return &ListPlayersResponse{
		Players: []*player.Player{},
	}, nil
}
